packages/discovery-react-components/src/components/SearchFacets/components/FieldFacets.tsx

```diff
@@ -39,65 +40,65 @@ export const FieldFacets: FC<FieldFacetsProps> = ({
   collapsedFacetsCount,
   onChange
 }) => {
-  const handleOnChange = (
-    selectedFacetName: string,
-    selectedFacetKey: string,
-    checked: boolean
-  ): void => {
+  const getFacetsForNameIndex = (selectedFacetName: string) => {
     const facetsForNameIndex = allFacets.findIndex(facet => {
       if (!facet.name) {
         return facet.field === selectedFacetName;
       }
       return facet.name === selectedFacetName;
     });
-    if (facetsForNameIndex > -1) {
-      const facetsForName = allFacets[facetsForNameIndex];
-      const multiselect = get(facetsForName, 'multiple_selections_allowed', true);
-      const facetResults: SelectableQueryTermAggregationResult[] = get(
-        facetsForName,
-        'results',
-        []
-      );
+    return facetsForNameIndex;
+  };
 
-      // Handle quirky case where toggling between checkboxes and radio buttons when
-      //   1. multiselect === false
-      //   2. going from 2 facets down to 1
-      const selectedFacets = filter(facetResults, ['selected', true]);
-      const usingRadioButtons = selectedFacets.length < 2 && !multiselect;
-      const selectedFacetResults: SelectableQueryTermAggregationResult[] = facetResults.map(
-        result => {
-          const key = get(result, 'key', '');
+  const handleOnChange = (selectedFacets: SelectedFacet[]): void => {
+    let updatedFacets = [...allFacets];
```

Comment:
This doesn't actually clone the objects in the array. It copies the references to them. So mutating any object in `updatedFacets` will also mutate the object in `allFacets`